src/main/java/request/HttpRequestReader.java

```diff
@@ -0,0 +1,13 @@
+package com.td.HttpServer;
+import java.io.IOException;
+
+public class HttpRequestReader implements IRequestReader {
```

Comment:
I suppose the `HttpRequestBuilder` could just use the `SocketIO` class directly.